third-party/realdds/include/realdds/topics/flexible/flexible-reader.h

```diff
@@ -0,0 +1,92 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+
+#include "flexible-msg.h"
+
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+
+#include <string>
+#include <memory>
+#include <chrono>
+#include <queue>
+#include <mutex>
+#include <condition_variable>
+
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+struct SubscriptionMatchedStatus;
+}
+}  // namespace fastdds
+}  // namespace eprosima
+
+
+namespace realdds {
+
+
+class dds_participant;
+class dds_topic;
+class dds_topic_reader;
+
+
+namespace topics {
+
+
+// Ease-of-use helper, easily taking care of basic use-cases that just want to easily read from some topic.
+// E.g.:
+//     flexible_reader topic( participant, "topic-name" );
+//     ...
+//     auto msg = topic.read().msg;
+//
+class flexible_reader
```

Comment:
Do we really need *flexible* reader/writer? Why won't `dds_topic_reader/writer` provide this capability?

Reply:
They do.
This is a higher-level class meant to ease use and hide many of the details.
We need the reader and writer in both sides of the POC (H and E). I.e., either copy the code or put it somewhere central.
These classes are totally optional but they do make things easier and simpler.

Reply:
They're also used in the unit-tests.

Reply:
`dds_topic_reader` doesn't have `wait_for_data` and dds_topic_writer` doesn't have `wait_for_readers`, but thinking about it again, I don't think it's a capability we will want for all topics, so it's OK to have a wrapper class that you will include only if you want this API.